sql/handler.cc

```diff
@@ -2627,6 +2629,8 @@ void **handler::ha_data(THD *thd) const
 
 THD *handler::ha_thd(void) const
 {
+  if (unlikely(cloned))
```

Comment:
I *really* dislike this part of the code. There is a lot of code all over the server that uses
file->in_use directly and the above 'change of thd' would get unnoticed.

It would be best if we could have this code such so that we never have to do a clone and never have a 'cloned' flag.